pkg/util/watch/listerwatcher_test.go

```diff
@@ -0,0 +1,895 @@
+			a: &client.ListOptions{
+				LabelSelector: nil,
+			},
+			b: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+			b: &client.ListOptions{
+				LabelSelector: nil,
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+			b: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+			b: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+			b: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+			b: &client.ListOptions{
+				LabelSelector: labels.Everything(),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.Nothing(),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+				),
+			},
+			b: &client.ListOptions{},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+				),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a:    &client.ListOptions{},
+			b: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+				),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+				),
+			},
+		},
+		{
+			name: "LabelSelectors",
+			a: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+				),
+			},
+			b: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+				),
+			},
+			expectedOpts: &client.ListOptions{
+				LabelSelector: labels.NewSelector().Add(
+					newLabelRequirement(t, "key", selection.Equals, []string{"value"}),
+					// TODO: add LabelSelector de-duping (minor optimization)
```

Comment:
Worth creating a ticket for?

Reply:
meh. it doesn't effect our usage. we only need it if we expand usage of these utility functions. like the fake clients, they're only as good as they need to be, because making them perfect is expensive and unnecessary. I mostly just put it in as a warning to anyone who runs into needing it later that it was intentionally left out for expedience.